sysdig/resource_sysdig_secure_cloud_auth_account.go

```diff
@@ -0,0 +1,174 @@
+package sysdig
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	draiosproto "github.com/draios/protorepo/cloudauth/go"
+	v2 "github.com/draios/terraform-provider-sysdig/sysdig/internal/client/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceSysdigSecureCloudauthAccount() *schema.Resource {
+	timeout := 5 * time.Minute
+
+	return &schema.Resource{
+		CreateContext: resourceSysdigSecureCloudauthAccountCreate,
+		UpdateContext: resourceSysdigSecureCloudauthAccountUpdate,
+		ReadContext:   resourceSysdigSecureCloudauthAccountRead,
+		DeleteContext: resourceSysdigSecureCloudauthAccountDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(timeout),
+			Update: schema.DefaultTimeout(timeout),
+			Read:   schema.DefaultTimeout(timeout),
+			Delete: schema.DefaultTimeout(timeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cloud_provider_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cloud_provider_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"aws", "gcp", "azure"}, false),
```

Comment:
I thought provider type is uint32 defined in the proto, check how it is done here https://github.com/sysdiglabs/terraform-provider-sysdig/pull/408/files#diff-58952c0c98afa44fb3f270b3577f925b263b05328de2b71cbddbb6bd1dea8229

Reply:
Updated!